ethapi/api.go

```diff
@@ -770,73 +769,51 @@ func (s *PublicBlockChainAPI) GetProof(ctx context.Context, address common.Addre
 	header, err := s.b.HeaderByNumber(ctx, number)
 	if header != nil && err == nil {
-		response := s.rpcMarshalHeader(header, s.calculateExtBlockApi(ctx, number))
-		if number == rpc.PendingBlockNumber {
-			// Pending header need to nil out a few fields
-			for _, field := range []string{"hash", "nonce", "miner"} {
-				response[field] = nil
-			}
-		}
+		response := s.rpcMarshalHeader(header, s.getBlockReceipts(ctx, number))
 		return response, err
 	}
 	return nil, err
 }
 
 // GetHeaderByHash returns the requested header by hash.
-func (s *PublicBlockChainAPI) GetHeaderByHash(ctx context.Context, hash common.Hash) map[string]interface{} {
+func (s *PublicBlockChainAPI) GetHeaderByHash(ctx context.Context, hash common.Hash) *evmcore.EvmHeaderJson {
 	header, _ := s.b.HeaderByHash(ctx, hash)
 	if header != nil {
-		return s.rpcMarshalHeader(header, s.calculateExtBlockApi(ctx, rpc.BlockNumber(header.Number.Uint64())))
+		return s.rpcMarshalHeader(header, s.getBlockReceipts(ctx, rpc.BlockNumber(header.Number.Uint64())))
 	}
 	return nil
 }
 
-func (s *PublicBlockChainAPI) calculateExtBlockApi(ctx context.Context, blkNumber rpc.BlockNumber) extBlockApi {
-	var ext extBlockApi
-	if s.b.CalcBlockExtApi() && blkNumber != rpc.EarliestBlockNumber {
-		receipts, err := s.b.GetReceiptsByNumber(ctx, blkNumber)
-		if err != nil {
-			return ext
-		}
-		if receipts.Len() != 0 {
-			ext.receiptsRoot = types.DeriveSha(receipts, trie.NewStackTrie(nil))
-			ext.bloom = types.CreateBloom(receipts)
-		} else {
-			ext.receiptsRoot = types.EmptyRootHash
-		}
+func (s *PublicBlockChainAPI) getBlockReceipts(ctx context.Context, blkNumber rpc.BlockNumber) types.Receipts {
+	if blkNumber == rpc.EarliestBlockNumber {
+		return nil
 	}
-	return ext
+	receipts, _ := s.b.GetReceiptsByNumber(ctx, blkNumber)
+	return receipts
 }
 
 // GetBlockByNumber returns the requested canonical block.
 //   - When blockNr is -1 the chain head is returned.
 //   - When blockNr is -2 the pending chain head is returned.
 //   - When fullTx is true all transactions in the block are returned, otherwise
 //     only the transaction hash is returned.
-func (s *PublicBlockChainAPI) GetBlockByNumber(ctx context.Context, number rpc.BlockNumber, fullTx bool) (map[string]interface{}, error) {
+func (s *PublicBlockChainAPI) GetBlockByNumber(ctx context.Context, number rpc.BlockNumber, fullTx bool) (*evmcore.EvmBlockJson, error) {
 	block, err := s.b.BlockByNumber(ctx, number)
 	if block != nil && err == nil {
-		response, err := s.rpcMarshalBlock(block, s.calculateExtBlockApi(ctx, number), true, fullTx)
-		if err == nil && number == rpc.PendingBlockNumber {
-			// Pending blocks need to nil out a few fields
-			for _, field := range []string{"hash", "nonce", "miner"} {
-				response[field] = nil
-			}
-		}
-		return response, err
+		return s.rpcMarshalBlock(block, s.getBlockReceipts(ctx, number), true, fullTx)
```

Comment:
In the old version there was a special case handling for pending blocks. Is this no longer needed?

This is also the case in other functions in this file.